sig/lrama/lexer.rbs

```diff
@@ -0,0 +1,84 @@
+module Lrama
+  class Lexer
+    type line_data = [String, Integer]
+    type reference = [::Symbol, Integer | String, Token[untyped]?, Integer, Integer]
+
+    class Type[SValue] < Struct[untyped]
```

Comment:
Do we need `SValue` type variable? Always `Type#id` is Interger and `Type#name` is String.

```
>> Lrama::Lexer::Token::P_expect
=> #<struct Lrama::Lexer::Type id=0, name="P_expect">
>> Lrama::Lexer::Token::Number
=> #<struct Lrama::Lexer::Type id=15, name="Number">
```

On the other hand. `Token` needs type variable because type of `s_value` depends on the value of `Type`.